pkg/api/tag.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+)
+
+func (s *server) getANewTag(w http.ResponseWriter, r *http.Request) {
+	tagName := mux.Vars(r)["name"]
+	if tagName == "" {
+		s.Logger.Debugf("bzz-tag: invalid tag name : %s", tagName)
+		s.Logger.Error("bzz-tag: invalid tag name")
+		jsonhttp.BadRequest(w, "invalid tag name")
+		return
+	}
+
+	tag, err := s.Tags.Create(tagName, 1, false)
+	if err != nil {
+		s.Logger.Debugf("bzz-chunk: tag creation error: %v", err)
+		s.Logger.Error("bzz-chunk: tag creation error")
+		jsonhttp.InternalServerError(w, "cannot create tag")
+		return
+	}
+	w.Header().Set("Content-Type", jsonhttp.DefaultContentTypeHeader)
+	w.Header().Set("Cache-Control", "no-cache, private, max-age=0")
+	jsonhttp.OK(w, tag)
```

Comment:
I think that the response here should be defined in this package and constructed from the tag, even if ti has the same fields. This would give clarity what is served by the api.